Menus/InventoryMenuGUI.h

```diff
@@ -0,0 +1,453 @@
+
+		/// <summary>
+		/// Gets whether the menu is enabled or not.
+		/// </summary>
+		/// <returns>Whether the menu is enabled.</returns>
+		bool IsEnabled() const { return m_EnabledState == EnabledState::Enabled || m_EnabledState == EnabledState::Enabling; }
+
+		/// <summary>
+		/// Gets whether the menu is enabled and not in carousel mode or not.
+		/// </summary>
+		/// <returns>Whether the menu is enabled and not in carousel mode</returns>
+		bool IsEnabledAndNotCarousel() const { return IsEnabled() && m_MenuMode != MenuMode::Carousel; }
+
+		/// <summary>
+		/// Gets whether the menu is in the process of enabling or disabling.
+		/// </summary>
+		/// <returns></returns>
+		bool IsEnablingOrDisabling() const { return m_EnabledState == EnabledState::Enabling || m_EnabledState == EnabledState::Disabling; }
+
+		/// <summary>
+		/// Gets whether the menu is at all visible or not.
+		/// </summary>
+		/// <returns>Whether the menu is visible.</returns>
+		bool IsVisible() const { return m_EnabledState != EnabledState::Disabled; }
+
+		/// <summary>
+		/// Enables or disables the menu and animates it in and out of view.
+		/// </summary>
+		/// <param name="enable">Whether to enable or disable the menu.</param>
+		void SetEnabled(bool enable);
+#pragma endregion
+
+#pragma region Full And Transfer Mode Getters and Setters
+		/// <summary>
+		/// Clears the selected item for this InventoryMenuGUI.
+		/// </summary>
+		void ClearSelectedItem();
+
+		/// <summary>
+		/// Sets the selected item for this InventoryMenuGUI.
+		/// If the button pointer or object pointer are null, or both indices are null, the selected item will instead be cleared.
+		/// </summary>
+		/// <param name="selectedItemButton">A pointer to the GUIButton that is selected.</param>
+		/// <param name="selectedItemObject">A pointer to the MovableObject that is selected.</param>
+		/// <param name="inventoryIndex">The index of this selected item in the displayed inventory. -1 means the item is not in the inventory.</param>
+		/// <param name="equippedItemIndex">The index of this selected item in the vector of equipped items. -1 means the item is not in the inventory.</param>
+		/// <param name="isBeingDragged">Whether or not the selected item is being dragged.</param>
+		void SetSelectedItem(GUIButton *selectedItemButton, MovableObject *selectedItemObject, int inventoryIndex, int equippedItemIndex, bool isBeingDragged);
+#pragma endregion
+
+#pragma region Updating
+		/// <summary>
+		/// Updates the state of this GUI each frame.
+		/// </summary>
+		virtual void Update();
+
+		/// <summary>
+		/// Draws the GUI.
+		/// </summary>
+		/// <param name="targetBitmap">A pointer to a BITMAP to draw on. Generally a screen BITMAP.</param>
+		/// <param name="targetPos">The absolute position of the target bitmap's upper left corner in the scene.</param>
+		virtual void Draw(BITMAP *targetBitmap, const Vector &targetPos = Vector()) const;
+#pragma endregion
+
+	private:
+		
```

Comment:
This is default vs behaviour for some reason. You're not wrong that it's ugleh though, so I'll fix it.

Reply:
It's not exactly default behavior.
If the lines are indented and you place the cursor on a blank line it will place it in line with the indented text, however it doesn't actually place spaces or tabs there until you start typing.
If you typed something and then erased it the inserted spaces/tabs remain there.

Reply:
It literally won't save it for me. If I delete the whitespace and hit save, vs ignores me and doesn't save that, so I can't commit it without either finding the specific vs formatting setting or opening the file in np++ or something dumb.

Reply:
Get the extension.
Just make sure you disable the auto-remove-trailing-spaces-on-save option from the settings.

Reply:
Extension seems like shit. It doesn't pick up the weird space after endregion but decides that my space after private is bad.
![image](https://user-images.githubusercontent.com/7739094/121837072-25c3e080-ccab-11eb-8114-cb8dc8e36da7.png)

Reply:
Bruh I literally explained how this shit works. 🧑‍⚖️
![image](https://user-images.githubusercontent.com/50018358/121837217-b9160980-ccdd-11eb-9815-38a6c3936ade.png)
It doesn't pick anything up after endregion BECAUSE THERE'S NOTHING THERE, unlike after private.